src/mlpack/core/util/arma_traits.hpp

```diff
@@ -217,6 +217,20 @@ struct GetDenseMatType<arma::SpMat<eT>>
   typedef arma::Mat<eT> type;
 };
 
+// Get the cube type corresponding to a given MatType.
+
+template<typename MatType>
+struct GetCubeType
+{
+  typedef arma::Cube<typename MatType::elem_type> type;
+};
+
+template<typename eT>
+struct GetCubeType<arma::Mat<eT>>
+{
+  typedef arma::Cube<eT> type;
+};
```

Comment:
Technically this second specialization shouldn't be necessary, since `type` will already be correct for any `arma::` type.

Secondarily, this will return weird results for things that aren't Armadillo types.  I wonder if it would be better to have `GetCubeType` fail for an unknown type---that is, don't specify a type for the default:

```
template<typename MatType>
struct GetCubeType;
```

And then have a version specialized for Armadillo:

```
template<typename MatType, typename = typename std::enable_if_t<IsArmaType<MatType>::value>>
struct GetCubeType<MatType>
{
  typedef arma::Cube<typename MatType::elem_type> type;
};
```

Or something like that.  We don't need to fix it here, but it's something I'm thinking about; the same issue applies to all of the other traits in this file.
```

Reply:
OKay, I will remove the second specialization for now and then we can figure out that later